pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -1,603 +0,0 @@
-    ):
-        # no metrics should be logged with graphs
-        if not enable_graph and isinstance(value, torch.Tensor):
-            value = value.detach()
-
-        if isinstance(value, torch.Tensor) and value.device.type == "xla":
-            value = value.cpu()
-
-        if 'meta' not in self:
-            self.__setitem__('meta', {})
-
-        # if user requests both step and epoch, then we split the metric in two automatically
-        # one will be logged per step. the other per epoch
-        was_forked = False
-        if on_step and on_epoch:
-            was_forked = True
-
-            # set step version
-            step_name = f'{name}_step'
-
-            self.__set_meta(
-                step_name,
-                value,
-                prog_bar,
-                logger,
-                on_step=True,
-                on_epoch=False,
-                reduce_fx=reduce_fx,
-                forked=False,
-                dataloader_idx=dataloader_idx,
-            )
-
-            self.__setitem__(step_name, value)
-
-            # set epoch version
-            epoch_name = f'{name}_epoch'
-
-            self.__set_meta(
-                epoch_name,
-                value,
-                prog_bar,
-                logger,
-                on_step=False,
-                on_epoch=True,
-                reduce_fx=reduce_fx,
-                forked=False,
-                dataloader_idx=dataloader_idx,
-            )
-            self.__setitem__(epoch_name, value)
-
-        # always log the original metric
-        self.__set_meta(
-            name,
-            value,
-            prog_bar,
-            logger,
-            on_step,
-            on_epoch,
-            reduce_fx,
-            forked=was_forked,
-            dataloader_idx=dataloader_idx,
-        )
-
-        # set the value
-        self.__setitem__(name, value)
-
-    def __set_meta(
-        self,
-        name: str,
-        value: Any,
-        prog_bar: bool,
-        logger: bool,
-        on_step: bool,
-        on_epoch: bool,
-        reduce_fx: Callable,
-        forked: bool,
-        dataloader_idx: Union[int, None],
-    ):
-        # set the meta for the item
-        meta_value = value
-        meta = dict(
-            prog_bar=prog_bar,
-            logger=logger,
-            on_step=on_step,
-            on_epoch=on_epoch,
-            reduce_fx=reduce_fx,
-            value=meta_value,
-            forked=forked,
-            dataloader_idx=dataloader_idx,
-        )
-
-        self['meta'][name] = meta
-
-        # track whether any input requires reduction on epoch end
-        _internal = self['meta']['_internal']
-        _internal['_reduce_on_epoch'] = max(_internal['_reduce_on_epoch'], on_epoch)
-
-    def track_batch_size(self, batch):
-        batch_size = Result.extract_batch_size(batch)
-        Result.attach_batch_size(batch_size, self)
-
-    @staticmethod
-    def extract_batch_size(batch):
```

Comment:
We have a few downstream projects which take a dependency on this utility function. 

In the short-term, we're replicating the logic within those projects. Is this the right long-term approach? Or are there plans to maintain and expose these utility functions? The new `ResultCollection` has a similar function (https://github.com/PyTorchLightning/pytorch-lightning/pull/7882/files#diff-bf4a466b765dd5c4128aca411d2999df7d8b6283723ea721be94713dac190112R444), but this is not static and requires a `Result` instance.

Reply:
I'd say it's better for you to replicate the function since it wasn't originally meant to be used for users.
But it's completely fair that we keep BC for it since it wasn't previously protected.

Do you want to open a PR so it becomes static again? Could also include some unit tests for it.